cmd/oidc-webhook-authenticator/app/options/options.go

```diff
@@ -5,20 +5,85 @@
 package options
 
 import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/spf13/pflag"
-	"k8s.io/apimachinery/pkg/runtime"
-	apiserver "k8s.io/apiserver/pkg/server"
-	genericoptions "k8s.io/apiserver/pkg/server/options"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
 )
 
 // Options contain the server options.
 type Options struct {
-	SecureServing  *genericoptions.SecureServingOptionsWithLoopback
-	Authentication *genericoptions.DelegatingAuthenticationOptions
-	Authorization  *genericoptions.DelegatingAuthorizationOptions
 	ResyncPeriod   resyncPeriod
+	ServingOptions ServingOptions
+}
+
+// ServingOptions are options applied to the authentication webhook server.
+type ServingOptions struct {
+	TLSCertFile                    string
+	TLSKeyFile                     string
+	ClientCAFile                   string
+	AuthenticationAlwaysAllowPaths []string
+
+	BindAddress string
+	BindPort    uint
+}
+
+// AddFlags adds server options to flagset
+func (s *ServingOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&s.TLSCertFile, "tls-cert-file", s.TLSCertFile, "File containing the x509 Certificate for HTTPS.")
+	fs.StringVar(&s.TLSKeyFile, "tls-private-key-file", s.TLSKeyFile, "File containing the x509 private key matching --tls-cert-file.")
+	fs.StringVar(&s.ClientCAFile, "client-ca-file", s.ClientCAFile, "If set, any request should present a client certificate signed by one of the authorities in the client-ca-file.")
+	fs.StringSliceVar(&s.AuthenticationAlwaysAllowPaths, "authentication-always-allow-paths", s.AuthenticationAlwaysAllowPaths, "A list of HTTP paths that do not require authentication. If authentication is not configured all paths are allowed.")
+
+	fs.StringVar(&s.BindAddress, "bind-address", "", "The IP address that the server will listen on. If unspecified all interfaces will be used.")
+	fs.UintVar(&s.BindPort, "bind-port", 10443, "The port that the server will listen on. If unspecified it will default to 10443.")
```

Comment:
```suggestion
	fs.StringVar(&s.BindAddress, "address", "", "The IP address that the server will listen on. If unspecified all interfaces will be used.")
	fs.UintVar(&s.BindPort, "port", 10443, "The port that the server will listen on. If unspecified it will default to 10443.")
```

Just `address` and `port` are sufficient for flag names. I find `bind` prefix confusing and was thinking it is related to https://en.wikipedia.org/wiki/BIND.

Reply:
It is related to the verb "bind"